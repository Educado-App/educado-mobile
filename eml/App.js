import React, { useState } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import LoginScreen from './screens/login/Login';
import RegisterScreen from './screens/register/Register';
import * as eva from '@eva-design/eva';
import { ApplicationProvider, IconRegistry } from '@ui-kitten/components';
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import ExerciseScreen from './screens/excercise/ExerciseScreen';
import { TailwindProvider } from 'tailwindcss-react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import SectionScreen from './screens/section/SectionScreen';
import { isFontsLoaded } from './constants/Fonts';
import LoadingScreen from './components/loading/Loading';
import WelcomeScreen from './screens/welcome/Welcome';
import CompleteSectionScreen from './screens/section/CompleteSection';
import NavBar from './components/navBar/NavBar';
import CompSwipeScreen from './screens/lectures/CompSwipeScreen';
import ErrorScreen from './screens/errors/ErrorScreen';
import CourseScreen from './screens/courses/CourseScreen';
import EditProfileScreen from './screens/profile/EditProfile';
import CertificateScreen from './screens/certificate/CertificateScreen';
import CompleteCourseScreen from './screens/courses/CompleteCourse';

const Stack = createNativeStackNavigator();

function WelcomeStack() {
	return (
		<Stack.Navigator initialRouteName={'Welcome'}>
			<Stack.Screen
				name="Welcome"
				component={WelcomeScreen}
				options={{
					headerShown: false,
				}}
			/>
		</Stack.Navigator>
	);
}

function LoginStack() {
	return (
		<Stack.Navigator initialRouteName={'Login'}>
			<Stack.Screen
				name="Login"
				component={LoginScreen}
				options={{
					headerShown: false,
				}}
			/>
			<Stack.Screen
				name="Register"
				component={RegisterScreen}
				options={{
					headerShown: false,
				}}
			/>
		</Stack.Navigator>
	);
}

function CertificateStack() {
	return (
		<Stack.Navigator initialRouteName={'Certificate'}>
			<Stack.Screen
				name="Certificate"
				component={CertificateScreen}
				options={{
					headerShown: false,
				}}
			/>
		</Stack.Navigator>
	);
}


function CourseStack() {
	return (
		<Stack.Navigator>
			<Stack.Screen
				name="Course"
				component={CourseScreen}
				options={{
					headerShown: false,
				}}
			/>
			<Stack.Screen
				name="CompleteSection"
				component={CompleteSectionScreen}
				options={{
					headerShown: false,
				}}
			/>
			<Stack.Screen
				name="Exercise"
				component={ExerciseScreen}
				options={{
					headerShown: false,
				}}
			/>
			<Stack.Screen
				name="Section"
				component={SectionScreen}
				initialParams={{ course_id: '' }}
				options={{ headerShown: false }}
			/>
			<Stack.Screen
				name="ErrorScreen"
				component={ErrorScreen}
				options={{
					headerShown: false,
				}}
			/>
		</Stack.Navigator>
	);
}

export function useWelcomeScreenLogic(loadingTime, onResult) {

	setTimeout(() => {
		const fetchData = async () => {
			try {
				const value = await AsyncStorage.getItem('hasShownWelcome');
				let initialRoute = 'WelcomeStack';
				let isLoading = true;

				if (value === 'true') {
					initialRoute = 'LoginStack';
				} else {
					await AsyncStorage.setItem('hasShownWelcome', 'true');
				}

				// Pass the results to the callback
				isLoading = false;
				onResult(initialRoute, isLoading);
			} catch (error) {
				console.error('Error retrieving or setting AsyncStorage data:', error);
			}
		};

		fetchData();
	}, loadingTime);

}


export default function App() {
	const fontsLoaded = isFontsLoaded();
	const [initialRoute, setInitialRoute] = useState('');
	const [isLoading, setIsLoading] = useState(true);

	// Callback function to handle the results
	const handleResult = (route, loading) => {
		setInitialRoute(route);
		setIsLoading(loading);
	};

	useWelcomeScreenLogic(3000, handleResult);

	// ************** Don't touch this code **************
	if (!fontsLoaded) {
		return null;
	}

	// Makes sure fonts are loaded before rendering the app
	if (isLoading && fontsLoaded) {
		return <LoadingScreen />;
	}
	// ***************************************************

	return (
		<TailwindProvider>
			<>
				<IconRegistry icons={EvaIconsPack} />
				<ApplicationProvider {...eva} theme={eva.light}>
					<NavigationContainer>
						<Stack.Navigator initialRouteName={initialRoute}>
							<Stack.Screen
								name="WelcomeStack"
								component={WelcomeStack}
								options={{ headerShown: false }}
							/>
							<Stack.Screen
								name="LoginStack"
								component={LoginStack}
								options={{ headerShown: false }}
							/>
							<Stack.Screen
								name="HomeStack"
								component={NavBar}
								options={{ headerShown: false }}
							/>
							<Stack.Screen
								name={'CourseStack'}
								component={CourseStack}
								options={{ headerShown: false }}
							/>
							<Stack.Screen
								name={'Section'}
								component={SectionScreen}
								initialParams={{ course_id: '' }}
								options={{ headerShown: false }}
							/>
							<Stack.Screen
								name={'CompleteSection'}
								component={CompleteSectionScreen}
								options={{ headerShown: false }}
							/>
							<Stack.Screen
								name="EditProfile"
								component={EditProfileScreen}
								options={{ headerShown: false }}
							/>
							<Stack.Screen
								name="Exercise"
								component={ExerciseScreen}
								options={{
									headerShown: false,
								}}
							/>
							<Stack.Screen
								name="Components"
								component={CompSwipeScreen}
								options={{
									headerShown: false,
								}}
							/>
							<Stack.Screen
								name="CertificateStack"
								component={CertificateStack}
								options={{
									headerShown: false,
								}}
							/>
							<Stack.Screen
								name="CompleteCourse"
								component={CompleteCourseScreen}
								options={{ headerShown: false }}
							/>
						</Stack.Navigator>
					</NavigationContainer>
				</ApplicationProvider>
			</>
		</TailwindProvider>
	);
}