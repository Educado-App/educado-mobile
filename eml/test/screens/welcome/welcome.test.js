import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from '../../../screens/welcome/Welcome';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { useWelcomeScreenLogic } from '../../../App'; // Update the import path as needed

let navigated = false;

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(() => { navigated = true; }),
  }),
}));

jest.mock('@react-navigation/bottom-tabs', () => ({
  createBottomTabNavigator: jest.fn(),
}));

jest.mock('@react-navigation/native-stack', () => ({
  createNativeStackNavigator: jest.fn(),
}));

let welcomeScreen;

beforeEach(() => {
  navigated = false;
  AsyncStorage.clear();
  welcomeScreen = renderer.create(<WelcomeScreen />);
});


afterEach(() => {
  welcomeScreen.unmount();
});

describe('WelcomeScreen', () => {
  it('renders welcomeScreen correctly', () => {
    expect(welcomeScreen.toJSON()).toMatchSnapshot();
  });

  it('should set initialRoute to LoginStack when hasShownWelcome is true', (done) => {
    jest.spyOn(AsyncStorage, 'getItem').mockResolvedValueOnce('true');

    const loadingTime = 1; 
    useWelcomeScreenLogic(loadingTime, (initialRoute, isLoading) => {
      expect(initialRoute).toBe('LoginStack');
      expect(isLoading).toBe(false);
      done(); // Finish the test
    });
  }); 

  it('should set initialRoute to WelcomeStack when hasShownWelcome is false', (done) => {
    jest.spyOn(AsyncStorage, 'getItem').mockResolvedValueOnce('false');

    const loadingTime = 1; 
    useWelcomeScreenLogic(loadingTime, (initialRoute, isLoading) => {
      expect(initialRoute).toBe('WelcomeStack');
      expect(isLoading).toBe(false);
      done(); // Finish the test
    });
  });
});