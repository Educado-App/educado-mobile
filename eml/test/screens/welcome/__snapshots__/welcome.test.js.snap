// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders welcomeScreen correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4291421676,
        4294967295,
      ],
      "endPoint": [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": [
        1,
        0.2,
      ],
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "display": "flex",
          },
          {
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "marginBottom": "20%",
            },
            {
              "paddingTop": "30%",
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            [
              {
                "width": 175.88,
              },
              {
                "height": 25.54,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "width": 750,
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "marginBottom": "15%",
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            [
              {
                "backgroundColor": "transparent",
                "flex": 1,
                "position": "relative",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            activeDotColor="#5ECCE9"
            activeDotStyle={
              {
                "height": 10,
                "width": 10,
              }
            }
            automaticallyAdjustContentInsets={false}
            autoplay={true}
            autoplayDirection={true}
            autoplayTimeout={10}
            bounces={false}
            contentContainerStyle={
              [
                {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            contentOffset={
              {
                "x": 0,
              }
            }
            disableNextButton={false}
            disablePrevButton={false}
            dotColor="#FFFFFF"
            dotStyle={
              {
                "height": 10,
                "width": 10,
              }
            }
            height={265}
            horizontal={true}
            index={0}
            loadMinimal={false}
            loadMinimalSize={1}
            loop={false}
            nextButton={
              <ForwardRef
                className="h-[25px] w-[25px] mr-4"
                component={[Function]}
              >
                <Path
                  d="M8.59003 17.1239L13.17 12.5439L8.59003 7.95385L10 6.54385L16 12.5439L10 18.5439L8.59003 17.1239Z"
                  fill="#383838"
                />
              </ForwardRef>
            }
            onIndexChanged={[Function]}
            onMomentumScrollEnd={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            pagingEnabled={true}
            prevButton={
              <ForwardRef
                className="h-[25px] w-[25px] ml-4"
                component={[Function]}
              >
                <Path
                  d="M15.41 17.1239L10.83 12.5439L15.41 7.95385L14 6.54385L8 12.5439L14 18.5439L15.41 17.1239Z"
                  fill="#383838"
                />
              </ForwardRef>
            }
            removeClippedSubviews={true}
            scrollEnabled={true}
            scrollsToTop={false}
            showsButtons={true}
            showsHorizontalScrollIndicator={false}
            showsPagination={true}
            showsVerticalScrollIndicator={false}
          >
            <View>
              <View
                style={
                  [
                    {
                      "height": 265,
                      "width": 750,
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "position": "relative",
                      },
                      {
                        "height": "100%",
                      },
                      {
                        "paddingLeft": 40,
                        "paddingRight": 40,
                      },
                      {
                        "alignItems": "center",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "top": 0,
                        },
                        {
                          "paddingLeft": 16,
                          "paddingRight": 16,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "Montserrat-Regular",
                          },
                          {
                            "color": "#383838",
                          },
                          {
                            "fontSize": 16,
                          },
                          [
                            {
                              "textAlign": "center",
                            },
                            {
                              "fontFamily": "Montserrat-Bold",
                            },
                            {
                              "fontSize": 24,
                            },
                          ],
                        ]
                      }
                    >
                      SEJA BEM-VINDO!
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        {
                          "bottom": 0,
                        },
                        {
                          "position": "absolute",
                        },
                        {
                          "paddingBottom": "27.5%",
                        },
                        {
                          "paddingLeft": 24,
                          "paddingRight": 24,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "Montserrat-Regular",
                          },
                          {
                            "color": "#383838",
                          },
                          {
                            "fontSize": 16,
                          },
                          [
                            {
                              "textAlign": "center",
                            },
                            {
                              "fontSize": 16,
                            },
                          ],
                        ]
                      }
                    >
                      Aqui, tornamos o aprendizado acessível e divertido para todos. Explore nossos conteúdos e comece sua jornada de desenvolvimento.
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "height": 265,
                      "width": 750,
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "position": "relative",
                      },
                      {
                        "height": "100%",
                      },
                      {
                        "paddingLeft": 40,
                        "paddingRight": 40,
                      },
                      {
                        "alignItems": "center",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "top": 0,
                        },
                        {
                          "paddingLeft": 16,
                          "paddingRight": 16,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "Montserrat-Regular",
                          },
                          {
                            "color": "#383838",
                          },
                          {
                            "fontSize": 16,
                          },
                          [
                            {
                              "textAlign": "center",
                            },
                            {
                              "fontFamily": "Montserrat-Bold",
                            },
                            {
                              "fontSize": 24,
                            },
                          ],
                        ]
                      }
                    >
                      FAÇA DOWNLOAD E ACESSE OFFLINE
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        {
                          "bottom": 0,
                        },
                        {
                          "position": "absolute",
                        },
                        {
                          "paddingBottom": "27.5%",
                        },
                        {
                          "paddingLeft": 24,
                          "paddingRight": 24,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "Montserrat-Regular",
                          },
                          {
                            "color": "#383838",
                          },
                          {
                            "fontSize": 16,
                          },
                          [
                            {
                              "textAlign": "center",
                            },
                            {
                              "fontSize": 16,
                            },
                          ],
                        ]
                      }
                    >
                      Este é o seu espaço para aprender de forma interativa e envolvente. Faça o download dos conteúdos e acesse offline quando quiser!
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "height": 265,
                      "width": 750,
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "position": "relative",
                      },
                      {
                        "height": "100%",
                      },
                      {
                        "paddingLeft": 40,
                        "paddingRight": 40,
                      },
                      {
                        "alignItems": "center",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "top": 0,
                        },
                        {
                          "paddingLeft": 16,
                          "paddingRight": 16,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "Montserrat-Regular",
                          },
                          {
                            "color": "#383838",
                          },
                          {
                            "fontSize": 16,
                          },
                          [
                            {
                              "textAlign": "center",
                            },
                            {
                              "fontFamily": "Montserrat-Bold",
                            },
                            {
                              "fontSize": 24,
                            },
                          ],
                        ]
                      }
                    >
                      CADASTRE-SE E EXPLORE
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        {
                          "bottom": 0,
                        },
                        {
                          "position": "absolute",
                        },
                        {
                          "paddingBottom": "27.5%",
                        },
                        {
                          "paddingLeft": 24,
                          "paddingRight": 24,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "Montserrat-Regular",
                          },
                          {
                            "color": "#383838",
                          },
                          {
                            "fontSize": 16,
                          },
                          [
                            {
                              "textAlign": "center",
                            },
                            {
                              "fontSize": 16,
                            },
                          ],
                        ]
                      }
                    >
                      Faça parte de nossa comunidade e descubra um mundo de aprendizado ao seu alcance, não importa sua formação acadêmica.
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            pointerEvents="none"
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 25,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#5ECCE9",
                    "borderRadius": 4,
                    "height": 8,
                    "marginBottom": 3,
                    "marginLeft": 3,
                    "marginRight": 3,
                    "marginTop": 3,
                    "width": 8,
                  },
                  {
                    "height": 10,
                    "width": 10,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 4,
                    "height": 8,
                    "marginBottom": 3,
                    "marginLeft": 3,
                    "marginRight": 3,
                    "marginTop": 3,
                    "width": 8,
                  },
                  {
                    "height": 10,
                    "width": 10,
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 4,
                    "height": 8,
                    "marginBottom": 3,
                    "marginLeft": 3,
                    "marginRight": 3,
                    "marginTop": 3,
                    "width": 8,
                  },
                  {
                    "height": 10,
                    "width": 10,
                  },
                ]
              }
            />
          </View>
          <View
            pointerEvents="box-none"
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "left": 0,
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                  "position": "absolute",
                  "top": 0,
                },
                {
                  "height": 265,
                  "width": 750,
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View />
            </View>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View>
                <RNSVGSvgView
                  bbHeight={25}
                  bbWidth={25}
                  focusable={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      [
                        {
                          "height": 25,
                        },
                        {
                          "width": 25,
                        },
                        {
                          "marginRight": 16,
                        },
                      ],
                      {
                        "flex": 0,
                        "height": 25,
                        "width": 25,
                      },
                    ]
                  }
                >
                  <RNSVGGroup>
                    <RNSVGPath
                      d="M8.59003 17.1239L13.17 12.5439L8.59003 7.95385L10 6.54385L16 12.5439L10 18.5439L8.59003 17.1239Z"
                      fill={4281874488}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "justifyContent": "space-around",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "paddingLeft": 24,
                "paddingRight": 24,
              },
              {
                "width": 750,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#5ECCE9",
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 40,
                "paddingRight": 40,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Montserrat-Regular",
                  },
                  {
                    "color": "#383838",
                  },
                  {
                    "fontSize": 16,
                  },
                  [
                    {
                      "textAlign": "center",
                    },
                    {
                      "fontFamily": "Montserrat-Bold",
                    },
                    {
                      "fontSize": 16,
                    },
                    {
                      "color": "#FFFFFF",
                    },
                  ],
                ]
              }
            >
              Entrar
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "marginTop": 24,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Montserrat-Regular",
                  },
                  {
                    "color": "#383838",
                  },
                  {
                    "fontSize": 16,
                  },
                  [
                    {
                      "textAlign": "center",
                    },
                    {
                      "fontFamily": "Montserrat-Bold",
                    },
                    {
                      "fontSize": 16,
                    },
                    {
                      "textDecorationLine": "underline",
                    },
                  ],
                ]
              }
            >
              Cadastrer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</ViewManagerAdapter_ExpoLinearGradient>
`;
